packages/js/test/plugins/auctionHouseModule/executeSale.test.ts

```diff
@@ -0,0 +1,409 @@
+    .executeSale({
+      listing,
+      bid,
+    })
+    .run();
+
+  // Then we created and returned the new Purchase
+  t.equal(purchase.asset.address.toBase58(), nft.address.toBase58());
+});
+
+test('[auctionHouseModule] it throws an error if Bid and Listing have different Auction House', async (t: Test) => {
+  // Given we have two Auction Houses and an NFT.
+  const sellerMx = await metaplex();
+  const buyerMx = await metaplex();
+
+  const nft = await createNft(sellerMx);
+
+  const { client: sellerClient } = await createAuctionHouse(sellerMx);
+  const { client: buyerClient } = await createAuctionHouse(buyerMx);
+
+  // And we listed that NFT for 1 SOL.
+  const { listing } = await sellerClient
+    .list({
+      mintAccount: nft.address,
+      price: sol(1),
+    })
+    .run();
+
+  // And we created a public bid on that NFT for 1 SOL but with different AH.
+  const { bid } = await buyerClient
+    .bid({
+      mintAccount: nft.address,
+      price: sol(1),
+    })
+    .run();
+
+  // When we execute a sale with given listing and bid.
+  const promise = sellerClient.executeSale({ listing, bid }).run();
+
+  // Then we expect an error.
+  await assertThrows(
+    t,
+    promise,
+    /You are trying to use Bid and Listing from different Auction Houses/
+  );
+});
+
+test('[auctionHouseModule] it throws an error if Bid and Listing have different Token', async (t: Test) => {
+  // Given we have an Auction House and two NFTs.
+  const mx = await metaplex();
+
+  const firstNft = await createNft(mx);
+  const secondNft = await createNft(mx);
+  const { client } = await createAuctionHouse(mx);
+
+  // And we listed that First NFT for 1 SOL.
+  const { listing } = await client
+    .list({
+      mintAccount: firstNft.address,
+      price: sol(1),
+    })
+    .run();
+
+  // And we created a public bid on that Second NFT.
+  const { bid } = await client
+    .bid({
+      mintAccount: secondNft.address,
+      price: sol(1),
+    })
+    .run();
+
+  // When we execute a sale with given listing and bid.
+  const promise = client.executeSale({ listing, bid }).run();
+
+  // Then we expect an error.
+  await assertThrows(
+    t,
+    promise,
+    /You are trying to execute a sale on a listing for a different NFT/
+  );
+});
+
+test('[auctionHouseModule] it executes sale on an Auction House with SPL treasury', async (t: Test) => {
+  // Given we have a Metaplex instance.
+  const mx = await metaplex();
+  const buyer = await createWallet(mx);
+
+  // And an existing SPL treasury.
+  const { token: treasuryToken } = await mx
+    .tokens()
+    .createTokenWithMint()
+    .run();
+
+  // And airdrop 10 tokens to buyer.
+  await mx
+    .tokens()
+    .createToken({
+      mint: treasuryToken.mint.address,
+      owner: buyer.publicKey,
+    })
+    .run();
+  await mx
+    .tokens()
+    .mint({
+      mint: treasuryToken.mint.address,
+      amount: token(10),
+      toOwner: buyer.publicKey,
+    })
+    .run();
```

Comment:
From my latest PR, using `tokens().mint(...)` will check if the associated token account exists and, if not, it will create it.

```suggestion
  await mx
    .tokens()
    .mint({
      mint: treasuryToken.mint.address,
      amount: token(10),
      toOwner: buyer.publicKey,
    })
    .run();
```